pkg/kubelet/kubelet.go

```diff
@@ -2964,8 +2980,17 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontaine
 			return
 		}
 		if kl.isPodResizeInProgress(allocatedPod, podStatus) {
-			// If a resize is in progress, make sure the cache has the correct state in case the Kubelet restarted.
-			kl.statusManager.SetPodResizeInProgressCondition(pod.UID, "", "", false)
+			// If a resize is in progress, make sure the cache has the correct state.
+			if len(kl.statusManager.GetPodResizeConditions(pod.UID)) == 2 {
+				// The new pod resize is either deferred or infeasible, which means if there is a pod
+				// resize in progress, it is associated with an older resize. In this case, the
+				// condition reflecting the older resize should also have the older observedGeneration.
+				gen := kl.statusManager.GetPodResizeInProgressCondition(pod.UID).ObservedGeneration
+				kl.statusManager.SetPodResizeInProgressCondition(pod.UID, "", "", gen, false)
+			} else {
```

Comment:
Is there any way to hit this condition if the new resize is pending? There would need to be a resize that was already allocated, but the ResizeInProgress condition wasn't added... I don't think that can happen? Just wanted to double check.